dashboard/src/app/dashboard/[dashboardId]/web-vitals/webVitalsTableSection.tsx

```diff
@@ -0,0 +1,200 @@
+'use client';
+
+import { use, useMemo, useState } from 'react';
+import { ColumnDef } from '@tanstack/react-table';
+import TabbedTable, { type TabDefinition } from '@/components/TabbedTable';
+import { DeviceIcon, BrowserIcon, OSIcon, FlagIcon } from '@/components/icons';
+import { formatCWV, getCwvStatusColor } from '@/utils/formatters';
+import MetricInfo from '@/app/dashboard/[dashboardId]/web-vitals/MetricInfo';
+import type { CoreWebVitalName } from '@/entities/webVitals';
+import type { fetchCoreWebVitalsByDimensionAction } from '@/app/actions/webVitals';
+import * as Flags from 'country-flag-icons/react/3x2';
+
+type Row = Awaited<ReturnType<typeof fetchCoreWebVitalsByDimensionAction>>[number];
+type DimRow = Awaited<ReturnType<typeof fetchCoreWebVitalsByDimensionAction>>[number];
+type Props = {
+  perPagePromise: Promise<Row[]>;
+  perDevicePromise: Promise<DimRow[]>;
+  perCountryPromise: Promise<DimRow[]>;
+  perBrowserPromise: Promise<DimRow[]>;
+  perOsPromise: Promise<DimRow[]>;
+};
+
+export default function WebVitalsTableSection({
+  perPagePromise,
+  perDevicePromise,
+  perCountryPromise,
+  perBrowserPromise,
+  perOsPromise,
+}: Props) {
+  const data = use(perPagePromise);
+  const devices = use(perDevicePromise);
+  const countries = use(perCountryPromise);
+  const browsers = use(perBrowserPromise);
+  const operatingSystems = use(perOsPromise);
+  type PercentileKey = 'p50' | 'p75' | 'p90' | 'p99';
+  const [activePercentile, setActivePercentile] = useState<PercentileKey>('p75');
+  const percentileIndex: Record<PercentileKey, number> = { p50: 0, p75: 1, p90: 2, p99: 3 };
+
+  const makeColumns = (label: string, renderIcon?: (key: string) => React.ReactNode): ColumnDef<Row>[] => {
```

Comment:
Suggestion: Wrap this in a useCallback, as it looks a little expensive to recompute 😄